tensorflow-frameworks/tensorflow-data/src/main/java/org/tensorflow/data/DatasetIterator.java

```diff
@@ -0,0 +1,214 @@
+package org.tensorflow.data;
+
+import org.tensorflow.DataType;
+import org.tensorflow.Graph;
+import org.tensorflow.Operand;
+import org.tensorflow.Output;
+import org.tensorflow.op.Op;
+import org.tensorflow.op.Ops;
+import org.tensorflow.tools.Shape;
+
+import java.util.List;
+
+/**
+ * Represents the state of an iteration through
+ * a tf.data Datset.
+ * <p>
```

Comment:
There should be an explicit notice in this javadoc that `DatasetIterator` is not a `java.util.Iterator`. It's unfortunate that we can't really change the name entirely, but I guess it's best to go with what the TF Python functions are called when they mirror the functionality.

Reply:
Good point, I added some explanation.